File: security/src/main/java/top/bettercode/summer/security/authorize/Anonymous.java
package top.bettercode.summer.security.authorize;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 匿名权限
 *
 * @author Peter Wu
 */
@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Inherited
@Documented
@ConfigAuthority("ROLE_ANONYMOUS")
public @interface Anonymous {

}

File: web/src/main/kotlin/top/bettercode/summer/security/authorize/Anonymous.kt
package top.bettercode.summer.security.authorize

import java.lang.annotation.Inherited

/**
 * 匿名权限
 *
 * @author Peter Wu
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Inherited
@MustBeDocumented
@ConfigAuthority(Anonymous.ROLE_ANONYMOUS_VALUE)
annotation class Anonymous {
    companion object {
        const val ROLE_ANONYMOUS_VALUE = "ROLE_ANONYMOUS"
    }
}
